server/src/main/java/org/elasticsearch/rest/RestRequest.java

```diff
@@ -62,7 +63,7 @@ public class RestRequest implements ToXContent.Params {
     private final HttpChannel httpChannel;
     private final ParsedMediaType parsedAccept;
     private final ParsedMediaType parsedContentType;
-    private final RestApiVersion restApiVersion;
+    private final Optional<RestApiVersion> restApiVersion;
```

Comment:
Is there a reason why this is kept as an optional field, and we do a `orElse(RestApiVersion.current());` every time? Can `RestApiVersion.current()` change, or could we resolve this at construction time and avoid the Optional field?

Reply:
Technically we can replace it with a nullable field instead of an `Optional`.

I didn't do that because I appreciate the support that the compiler provides in avoiding accidental access to a potentially null value, but preferences will vary.

Reply:
I was thinking about the 2 fields, but the difference is indeed minimal and I think a matter of preference. Thanks for looking into this though!